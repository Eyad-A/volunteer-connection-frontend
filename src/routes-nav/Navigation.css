.Navigation {
    background-color: #264653;
    border-bottom: 1px solid rgb(0, 0, 0);
    padding-left: 20px;
    padding-right: 50px;
    font-size: 18px;
}

.dropdown-menu {
    background-color: #264653;
}

.navbar-brand {
    font-size: 18px;
    color: #fff !important;
}

.nav-link {
    color: #fff !important;
}

.nav-link:hover {
    color: #FFC107 !important;
}

