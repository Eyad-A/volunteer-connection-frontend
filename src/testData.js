import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
    username: "demouser",
    first_name: "demouser",
    last_name: "demouser",
    email: "demouser@demouser.com",
    skill: "Developer"
};

const UserProvider = ({
    children, currentUser = demoUser, hasConnectedToCompany = () => false 
}) => (
    <UserContext.Provider value={{ currentUser, hasConnectedToCompany }}>
        {children}
    </UserContext.Provider>
);

export { UserProvider};