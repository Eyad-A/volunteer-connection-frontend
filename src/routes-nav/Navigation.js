import React, { useContext } from "react";
import { Link, NavLink } from "react-router-dom";
import UserContext from "../auth/UserContext";
import "./Navigation.css"

/**
 * Nav bar that shows up on top of every page
 * If the user/company is logged in, will show links to 
 * companies/profile. Otherwise, will show 
 * login/signup links. 
 */

function Navigation({ logout }) {

  const { currentUser } = useContext(UserContext);

  function loggedInNav() {
    return (
      <div className="container-fluid">
        <Link className="navbar-brand link-dark" to="/">
          Volunteer Connection
        </Link>
        <div className="justify-content-end">
          <ul className="navbar-nav ml-auto justify-content-end">
            <li className="nav-item mr-4">
              <NavLink className="nav-link" to="/companies">
                Companies
              </NavLink>
            </li>
            <li className="nav-item mr-4">
              <NavLink className="nav-link" to="/user-connections">
                Connections 
              </NavLink>
            </li>
            <li className="nav-item mr-4">
              <NavLink className="nav-link" to="/profile-user">
                Profile
              </NavLink>
            </li>
            <li className="nav-item mr-4">
              <NavLink className="nav-link" to="/" onClick={logout}>
                Logout
              </NavLink>
            </li>
          </ul>
        </div>

      </div>

    );
  }

  function loggedOutNav() {
    return (
      <div className="container-fluid">
        <Link className="navbar-brand link-dark" to="/">
          Volunteer Connection 
        </Link>
        <div className="justify-content-end">
          <ul className="navbar-nav ml-auto justify-content-end">
            <li className="nav-item mr-4">
              <NavLink className="nav-link" to="/companies">
                Companies
              </NavLink>
            </li>            
            <li className="navbar-item mr-4">
              <NavLink className="nav-link" to="/login-user">
                Login
              </NavLink>
            </li>
            <li className="navbar-item mr-4">
              <NavLink className="nav-link" to="/signup-user">
                Signup
              </NavLink>
            </li>
          </ul>
        </div>

      </div>
    );
  }

  return (
    <nav className="Navigation navbar navbar-expand-md">
      {currentUser ? loggedInNav() : loggedOutNav()}
    </nav>
  );
}

export default Navigation;