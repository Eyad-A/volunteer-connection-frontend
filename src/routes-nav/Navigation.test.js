import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Navigation from "./Navigation";
import { UserProvider } from "../testData";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <Navigation />
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider>
            <Navigation />
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

