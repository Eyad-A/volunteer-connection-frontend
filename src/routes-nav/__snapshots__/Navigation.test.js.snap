// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand link-dark"
        href="/"
      >
        Volunteer Connection
      </a>
      <div
        class="justify-content-end"
      >
        <ul
          class="navbar-nav ml-auto justify-content-end"
        >
          <li
            class="nav-item mr-4"
          >
            <a
              class="nav-link"
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item mr-4"
          >
            <a
              class="nav-link"
              href="/user-connections"
            >
              Connections
            </a>
          </li>
          <li
            class="nav-item mr-4"
          >
            <a
              class="nav-link"
              href="/profile-user"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item mr-4"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
